### ISSUE TITLE
When erroring on inadequate alignment, consider reminding reader that packed can lower alignment

### ISSUE THREAD

**Issue by pnkfelix** [rust-lang/rust#110199]
### Code

```Rust
#[repr(packed)]
pub struct Foo {
    a: usize,
    b: [u16; 1],
}

pub fn foo(f: &Foo) {
    bar(f.b.as_ptr());
}

pub fn bar(_ptr: *const u16) {}

#[repr(C, align(8))]
pub struct ByteThenFoo { a: u8, b: Foo }

fn main() {
    let mut v = vec![];
    v.push(ByteThenFoo {
        a: 1,
        b: Foo { a: 123, b: [2] },
    });
    
    let x = v.last().unwrap();
    foo(&x.b);
}
```

### Current output

```Shell
error[E0793]: reference to packed field is unaligned
 --> src/main.rs:8:9
  |
8 |     bar(f.b.as_ptr());
  |         ^^^^^^^^^^^^
  |
  = note: fields of packed structs are not properly aligned, and creating a misaligned reference is undefined behavior (even if that reference is never dereferenced)
  = help: copy the field contents to a local variable, or replace the reference with a raw pointer and use `read_unaligned`/`write_unaligned` (loads and stores via `*p` must be properly aligned even when using raw pointers)

For more information about this error, try `rustc --explain E0793`.
error: could not compile `playground` (bin "playground") due to previous error
```

### Desired output

much like the above, but point out that `#[packed]` makes the *struct itself* have lower alignment

### Rationale and extra context

Spawned off of https://github.com/rust-lang/rust/issues/109745#issuecomment-1497014963

Its not intuitive to many that `#[packed]` will imply a lower `#[align]`ment for the whole struct.

**Comment by Nilstrieb** [comment]
Mentor instructions: find where the error is emitted (using grep or -Ztreat-err-as-bug). Then, add the note to the diagnostics struct.
See https://rustc-dev-guide.rust-lang.org/diagnostics.html for more information.

**Comment by gernot-ohner** [comment]
Is this issue solved by PR https://github.com/rust-lang/rust/pull/110973?

**Comment by bindsdev** [comment]
> Is this issue solved by PR #110973?

It seems like it was part of the 1.71 milestone. I'm not sure why this issue wasn't closed.

**Comment by Dylan-DPC** [comment]
Closing as this is fixed.
